share/functions/__fish_config_interactive.fish

```diff
@@ -241,17 +241,26 @@ function __fish_config_interactive -d "Initializations that should be performed
         # __fish_enable_focus
     end
 
-    function __fish_winch_handler --on-signal WINCH -d "Repaint screen when window changes size"
-        # VTE reflows the text itself, so us doing it inevitably races against it.
-        # Guidance from the VTE developers is to let them repaint.
-        if set -q VTE_VERSION
-            return
-        end
+    # VTE reflows the text itself, so us doing it inevitably races against it.
+    # Guidance from the VTE developers is to let them repaint.
+    if set -q VTE_VERSION
         # Same for alacritty
-        if string match -q -- 'alacritty*' $TERM
-            return
+        or string match -q -- 'alacritty*' $TERM
+        set -g fish_handle_reflow 0
+    else if set -q KONSOLE_VERSION
+        and test "$KONSOLE_VERSION" -ge 210400 2>/dev/null
+        # Konsole since version 21.04(.00)
+        # Note that this is optional, but since we have no way of detecting it
+        # we go with the default, which is true.
+        set -g fish_handle_reflow 0
+    else if not set -q fish_handle_reflow
+        set -g fish_handle_reflow 1
```

Comment:
Reminder for myself: This needs to not touch $fish_handle_reflow *at all* if it is set - or we'd override customization from config.fish.